### PR TITLE
Implement "Default Post Language" and "Ask Before"-Alerts

### PR DESCRIPTION
**Pull request by kimar**
# Implement "Default Post Language" and "Ask Before"-Alerts

## Demo

| Default Post Language | Alte Text Missing |
|---|---|
| RocketSim_Recording_iPhone_15_6 1_2024-02-26_14 35 07 |RocketSim_Recording_iPhone_15_6 1_2024-02-26_14 36 10|

## Acceptance Criteria

* Defaults
  * Since these are new settings in this release, we don't have to worry about prior user preferences. Everything is on by default (unless any of these are synced back to the server. In which case, use the server's value.)

* Default post language
  * If this is unset, default to the system language. If it's out of view, scroll to it.

* Add alt text reminders
* Ask before unfollowing someone (maps to user_setting_unfollow_modal)
* Ask before boosting
* Ask before deleting posts

### CODE REVIEW

File: Mastodon/Scene/Settings/General Settings/GeneralSettingsViewController.swift [hunk 1]
```diff
@@ -27,7 +35,7 @@ class GeneralSettingsViewController: UIViewController {
 
     let sections: [GeneralSettingsSection]
 
-    init(setting: Setting) {
+    init(appContext: AppContext, setting: Setting) async {
```
Comment (zeitschlag) [inline Mastodon/Scene/Settings/General Settings/GeneralSettingsViewController.swift]: Why do we need an `async` here?

File: Mastodon/Scene/Settings/SettingsCoordinator.swift [hunk 2]
```diff
@@ -57,10 +58,15 @@ extension SettingsCoordinator: SettingsViewControllerDelegate {
     func didSelect(_ viewController: UIViewController, entry: SettingsEntry) {
         switch entry {
             case .general:
-                let generalSettingsViewController = GeneralSettingsViewController(setting: setting)
+            Task { @ MainActor in
+                sceneCoordinator.showLoading()
```
Comment (zeitschlag) [inline Mastodon/Scene/Settings/SettingsCoordinator.swift]: Does showing the General Settings take so long that we need to show the spinner?

File: Mastodon/Supporting Files/AppDelegate.swift [hunk 3]
```diff
@@ -12,7 +12,7 @@ import MastodonCore
 import MastodonUI
 
 @main
-class AppDelegate: UIResponder, UIApplicationDelegate {
+ class AppDelegate: UIResponder, UIApplicationDelegate {
```
Comment (zeitschlag) [inline Mastodon/Supporting Files/AppDelegate.swift]:
```suggestion
class AppDelegate: UIResponder, UIApplicationDelegate {
```

File: Mastodon/Scene/Settings/General Settings/GeneralSettingSelectionCell.swift [hunk 4]
```diff
@@ -45,5 +50,17 @@ class GeneralSettingSelectionCell: UITableViewCell {
...
         contentConfiguration = content
     }
+    
+    private func configureLanguageSetting(_ setting: GeneralSetting.Language, viewModel: GeneralSettingsViewModel) {
+        tintColor = Asset.Colors.Brand.blurple.color
+        accessoryType = .none
+        
+        var content = defaultContentConfiguration()
+        content.text = setting.title
+        content.secondaryText = LanguagePicker.availableLanguages().first(where: { $0.localeId == UserDefaults.shared.defaultPostLanguage })?.exonym
```
Comment (zeitschlag) [inline Mastodon/Scene/Settings/General Settings/GeneralSettingSelectionCell.swift]: The smaller text looked weird, so I checked Figma. According to Figma, it should look like this:

Screenshot 2024-02-28 at 09 08 05

Currently it looks like this:

Screenshot 2024-02-28 at 09 08 13

Maybe you can have another look? IIRC it's somethingsomething `UIListContentConfiguration.valueCell()` (but I'm not sure TBH

File: MastodonSDK/Sources/MastodonUI/Scene/ComposeContent/Toolbar/LanguagePicker.swift [hunk 5]
```diff
@@ -36,31 +75,42 @@ struct LanguagePicker: View {
...
+                .id(lang.id)
             }
             .listStyle(.plain)
             .searchable(text: $query, placement: .navigationBarDrawer(displayMode: .always))
-            .navigationTitle(L10n.Scene.Compose.Language.title)
-            .navigationBarTitleDisplayMode(.inline)
-        }.navigationViewStyle(.stack)
+            .onAppear {
+                filteredLanguages = query.isEmpty ? languages : languages.filter { $0.contains(query) }
+                DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(200)) { // when scrolling to quickly it'll overlap with other drawcycles and mess up the position :-(
```
Comment (zeitschlag) [inline MastodonSDK/Sources/MastodonUI/Scene/ComposeContent/Toolbar/LanguagePicker.swift:100]: To it felt weird that the search bar appears without an animation. But I don't know how to improve this. Do you have an idea? (Feel free to just mark this as resolved)

https://github.com/mastodon/mastodon-ios/assets/2580019/1e0f4441-94b0-467b-a45d-568092c0eca8

**Review by zeitschlag — APPROVED** [review]
LGTM! Thank you for your work.

I added some feedback here and there, maybe you can have a look at that? If it'd be too much work, just mark them as resolved. I don't see any real blockers

File: Mastodon/Scene/Settings/General Settings/GeneralSettingsViewController.swift (hunk 1 above)
Comment (DK1DA) [reply Mastodon/Scene/Settings/General Settings/GeneralSettingsViewController.swift]: Good catch! That was a left-over of something I tried out and can certainly be removed, I'M on it 👍

File: Mastodon/Scene/Settings/SettingsCoordinator.swift (hunk 2 above)
Comment (DK1DA) [reply Mastodon/Scene/Settings/SettingsCoordinator.swift]: Aaand you found another left-over, sorrythis whole chunk of code can be refactored. I'll do it now.

File: MastodonSDK/Sources/MastodonUI/Scene/ComposeContent/Toolbar/LanguagePicker.swift (hunk 5 above)
Comment (DK1DA) [reply MastodonSDK/Sources/MastodonUI/Scene/ComposeContent/Toolbar/LanguagePicker.swift:100]: I've tried a lot of things, but unfortunately not found any good solution to fix this. It's due to the `.searchable(:)` viewModifier used in `LanguagePicker` and the fact that we're pushing the SwiftUI view insdie a UIHostingController onto a UIKit NavigationStack here.

Comment (DK1DA) [reply MastodonSDK/Sources/MastodonUI/Scene/ComposeContent/Toolbar/LanguagePicker.swift:100]: I'll merge as-is for now.

**Pull request merged**